falcon/testing/client.py

```diff
@@ -685,4 +686,8 @@ def simulate_request(self, *args, **kwargs):
 
             kwargs['headers'] = merged_headers
 
+        if six.PY2 and 'headers' in kwargs:
+            for key, value in six.iteritems(kwargs['headers']):
+                kwargs['headers'][key] = str(value) if type(value) == six.text_type else value
```

Comment:
It is more idiomatic to use ``isinstance(...)`` than comparing types this way.

Furthermore you are targeting ``PY2`` specifically, so you could just say ``unicode`` directly. The latter remark is a matter of taste, not a requirement though. And performance is unimportant in testing client.